pkg/v7/configmap/desired.go

```diff
@@ -69,20 +69,31 @@ func (s *Service) GetDesiredState(ctx context.Context, clusterConfig ClusterConf
 	return desiredConfigMaps, nil
 }
 
-func (s *Service) checkHelmReleaseExists(ctx context.Context, releaseName string, clusterConfig ClusterConfig) (bool, error) {
-	tenantHelmClient, err := s.newTenantHelmClient(ctx, clusterConfig)
+// hasLegacyIngressController checks if the Ingress Controller deployment
+// exists and was created via k8scloudconfig. If so the chart migration
+// logic must be enabled.
+func (s *Service) hasLegacyIngressController(ctx context.Context, releaseName string, clusterConfig ClusterConfig) (bool, error) {
+	tenantK8sClient, err := s.newTenantK8sClient(ctx, clusterConfig)
 	if err != nil {
 		return false, microerror.Mask(err)
 	}
 
-	_, err = tenantHelmClient.GetReleaseContent(releaseName)
-	if helmclient.IsReleaseNotFound(err) {
+	ingressControllerDeploy, err := tenantK8sClient.Extensions().Deployments(metav1.NamespaceSystem).Get("nginx-ingress-controller", metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		// No deployment. So nothing to migrate.
 		return false, nil
 	} else if err != nil {
 		return false, microerror.Mask(err)
 	}
 
-	return true, nil
+	// ServiceType label is only present on deployments created via
+	// chart-operator.
+	serviceType, ok := ingressControllerDeploy.Labels[label.ServiceType]
+	if !ok || serviceType != label.ServiceTypeManaged {
+		return true, nil
+	}
```

Comment:
We use absence of the service type label to detect the deployment was created via `k8scloudconfig`. This is the same label used by the migration jobs.